tools/codegen/src/main.rs

```diff
@@ -130,6 +152,66 @@ fn main() -> Result<()> {
             Err(e) => eprintln!("failed to load old manifest: {e}"),
         }
     }
+
+    // Populate website
+    manifests.website = None;
+    if let Some(website) = base_info.website {
+        if website.is_empty() || website == base_info.repository {
+            panic!("Please do not put the repository in website, or set website to an empty value");
+        }
+        eprintln!("set to {website}");
+        manifests.website = Some(website);
+    } else {
+        if let Some(ref crate_io_info) = crates_io_info {
+            if let Some(ref homepage) = crate_io_info.crate_.homepage {
+                if !homepage.is_empty()
+                    && homepage != &base_info.repository
+                    && !homepage.contains("crates.io")
+                {
+                    manifests.website = Some(homepage.clone());
+                    eprintln!("set to {homepage}");
+                }
+            }
+        }
+        if manifests.website.is_none() {
+            if let Some(homepage) = repo_info.homepage {
+                if !homepage.is_empty()
+                    && homepage != base_info.repository
+                    && !homepage.contains("crates.io")
+                {
+                    manifests.website = Some(homepage.clone());
+                    eprintln!("set to {homepage}");
+                }
+            }
+        }
+    }
+
+    // Populate license_markdown
+    if let Some(license_markdown) = base_info.license_markdown {
+        if license_markdown.is_empty() {
+            panic!("license_markdown can not be an empty value");
+        }
+        manifests.license_markdown = license_markdown;
+    } else if let Some(detail) = crates_io_version_detail {
+        if let Some(license) = detail.license {
+            if let Some(license_markdown) =
+                get_license_markdown(&license, &repo.to_string(), &repo_info.default_branch)
+            {
+                manifests.license_markdown = license_markdown;
+            }
+        } else {
+            panic!("No license found in crate metadata; set license_markdown manually")
```

Comment:
revise slightly to indicate: license markdown couldn't be created from crate metadata